sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -3481,11 +3488,60 @@ class AstBuilder extends DataTypeAstBuilder
   /**
    * Create a generation expression string.
    */
-  override def visitGenerationExpression(ctx: GenerationExpressionContext): String =
+  override def visitGeneratedColumn(ctx: GeneratedColumnContext): String =
     withOrigin(ctx) {
       getDefaultExpression(ctx.expression(), "GENERATED").originalSQL
     }
 
+  /**
+   * Parse and verify IDENTITY column definition.
+   *
+   * @param ctx      The parser context.
+   * @param dataType The data type of column defined as IDENTITY column. Used for verification.
+   * @return Tuple containing start, step and allowExplicitInsert.
+   */
+  protected def parseIdentityColumn(
+                                     ctx: IdentityColumnContext,
+                                     dataType: DataType): IdentityColumnSpec = {
+    if (dataType != LongType) {
+      throw QueryParsingErrors.identityColumnUnsupportedDataType(ctx, dataType.toString)
+    }
```

Comment:
Maybe this should be a decision for the underlying framework. Theoretically, a framework extending Spark might want to support INT type identity column. So I think we can do two things:
1. Delegate this decision to the underlying framework by offering an interface they can override (do we have example of such code in the codebase?) 
2. Allowlist every valid type for identity column, knowing that the `step` and `start` are defined as `integers`. Let the underlying framework that extends Spark throw its own error if it doesn't want to support more than LongType.